packages/ai-chat-ui/src/browser/chat-view-widget.tsx

```diff
@@ -176,9 +176,9 @@ export class ChatViewWidget extends BaseWidget implements ExtractableWidget, Sta
     }
 
     protected onCancel(requestModel: ChatRequestModel): void {
-        // TODO we should pass a cancellation token with the request (or retrieve one from the request invocation) so we can cleanly cancel here
-        // For now we cancel manually via casting
-        (requestModel as ChatRequestModelImpl).response.cancel();
+        if (requestModel instanceof ChatRequestModelImpl) {
+            requestModel.cancel();
+        }
```

Comment:
As we're already cleaning up the code:
We should avoid this cast wherever possible, ideally everywhere. Having to use the cast means that there is something not properly cleaned up in the code base. Either:
- the `chat-view-widget` should have access to the `ChatRequestModelImpl` directly (without casting), OR
- the cancel should be triggered differently, e.g. via the token or a callback from the invocation

Generally: Should have the `chat-view-widget` access to the `ChatRequestModelImpl`? Only mutators should have access.

Reply:
Right, avoiding those casts should indeed be the goal!
I've moved the cancellation now into the chat service, as the widget already interacts with it to send the request and the chat service already deals with the instantiation and its concrete impl classes.
WDYT?